builder/vsphere/common/step_export.go

```diff
@@ -0,0 +1,295 @@
+//
+// Example usage:
+//
+// ```json
+// ...
+//   "vm_name": "example-ubuntu",
+// ...
+//   "export": {
+//     "force": true,
+//     "output_directory": "./output_vsphere"
+//   },
+// ```
+// The above configuration would create the following files:
+//
+// ```
+// ./output_vsphere/example-ubuntu-disk-0.vmdk
+// ./output_vsphere/example-ubuntu.ovf
+// ```
+type ExportConfig struct {
+	// name of the ovf. defaults to the name of the VM
+	Name string `mapstructure:"name"`
+	// overwrite ovf if it exists
+	Force bool `mapstructure:"force"`
+	// include iso and img image files that are attached to the VM
+	Images bool `mapstructure:"images"`
+	// generate manifest using SHA 1, 256, 512. use 0 (default) for no manifest
+	Sha       int          `mapstructure:"sha"`
+	OutputDir OutputConfig `mapstructure:",squash"`
+	// Advanced ovf export options. Options can include:
+	// * mac - MAC address is exported for all ethernet devices
+	// * uuid - UUID is exported for all virtual machines
+	// * extraconfig - all extra configuration options are exported for a virtual machine
+	// * nodevicesubtypes - resource subtypes for CD/DVD drives, floppy drives, and serial and parallel ports are not exported
+	//
+	// For example, adding the following export config option would output the mac addresses for all Ethernet devices in the ovf file:
+
+	// ```json
+	// ...
+	//   "export": {
+	//     "options": ["mac"]
+	//   },
+	// ```
+	Options []string `mapstructure:"options"`
+}
+
+var sha = map[int]func() hash.Hash{
+	1:   sha1.New,
+	256: sha256.New,
+	512: sha512.New,
+}
+
+func (c *ExportConfig) Prepare(ctx *interpolate.Context, lc *LocationConfig, pc *common.PackerConfig) []error {
+	var errs *packer.MultiError
+
+	errs = packer.MultiErrorAppend(errs, c.OutputDir.Prepare(ctx, pc)...)
+
+	if _, ok := sha[c.Sha]; !ok {
+		errs = packer.MultiErrorAppend(errs, fmt.Errorf("unknown hash: sha%d", c.Sha))
+	}
+
+	if c.Name == "" {
+		c.Name = lc.VMName
+	}
+	target := getTarget(c.OutputDir.OutputDir, c.Name)
+	if !c.Force {
+		if _, err := os.Stat(target); err == nil {
+			errs = packer.MultiErrorAppend(errs, fmt.Errorf("file already exists: %s", target))
+		}
+	}
+
+	if err := os.MkdirAll(c.OutputDir.OutputDir, 0750); err != nil {
+		errs = packer.MultiErrorAppend(errs, errors.Wrap(err, "unable to make directory for export"))
+	}
+
+	if errs != nil && len(errs.Errors) > 0 {
+		return errs.Errors
+	}
+
+	return nil
+}
+
+func getTarget(dir string, name string) string {
+	return filepath.Join(dir, name+".ovf")
+}
+
+type StepExport struct {
+	Name      string
+	Force     bool
+	Images    bool
+	Sha       int
+	OutputDir string
+	Options   []string
+	mf        bytes.Buffer
+}
+
+func (s *StepExport) Cleanup(multistep.StateBag) {
+}
+
+func (s *StepExport) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	ui := state.Get("ui").(packer.Ui)
+	vm := state.Get("vm").(*driver.VirtualMachine)
+
+	lease, err := vm.Export()
```

Comment:
In testing there was no output indicating that the VM was being exported. Seeing as it will take some time to export it would good to display a message to the user indicating that the Export has begun, in the download step maybe it can output what is being downloaded, and finally end with a message that the exported created the ovf file. 

This is what my current output looks like 

```
==> vsphere-iso: Shut down VM...
==> vsphere-iso: Deleting Floppy drives...
==> vsphere-iso: Deleting Floppy image...
==> vsphere-iso: Eject CD-ROM drives...
    vsphere-iso: exporting file: example-ubuntu-disk-0.vmdk
==> vsphere-iso: Clear boot order...
Build 'vsphere-iso' finished.
```

Reply:
added some messages to let the user know what was happening

Reply:
Nice! :100: 

```
==> vsphere-iso: Shut down VM...
==> vsphere-iso: Deleting Floppy drives...
==> vsphere-iso: Deleting Floppy image...
==> vsphere-iso: Eject CD-ROM drives...
    vsphere-iso: Starting export...
    vsphere-iso: Downloading: ubuntu-disk-0.vmdk
    vsphere-iso: Exporting file: ubuntu-disk-0.vmdk
    vsphere-iso: Writing ovf...
==> vsphere-iso: Clear boot order...
Build 'vsphere-iso' finished.
```